app/services/queries/reviews_turnaround_per_user_project_query.rb

```diff
@@ -0,0 +1,32 @@
+module Queries
+  class ReviewsTurnaroundPerUserProjectQuery < BaseService
+    attr_reader :time_interval
+
+    def initialize(time_interval:)
+      @time_interval = time_interval
+    end
+
+    def call
+      run
+    end
+
+    private
+
+    def query
+      Events::Review.joins(:review_request, pull_request: :project).includes(:review_request)
+                    .where(opened_at: @time_interval.starting_at..@time_interval.ending_at)
+    end
+
+    def run
+      query.collect do |review|
+        { entity: find_user_project(review.owner, review.pull_request.project),
```

Comment:
We have an N+1 here, but I'm not sure how to fix it 🤔

Reply:
I set a breakpoint before the `run` method, and put `ActiveRecord::Base.logger = Logger.new(STDOUT)` to get the output of the query and I got just one query.

Reply:
but if you are fetching the owner for every record returned, how can it not be triggering N queries?

Reply:
By doing the join I am getting all the owners. What I am missing is the owners in the join.